### ISSUE TITLE
The CER of  librispeech? 

### ISSUE THREAD

**Issue by flp1990** [gentaiscool/end2end-asr-pytorch#39]
hello, the best cer in librispeech test-clean is ?

**Comment by gentaiscool** [comment]
hi @flp1990 that depends to the architecture you use. You may check #34. There are some results reported there. I think, you can get better results by carefully tuning the model (+ deeper + wider models + regularization, and larger beam search).

**Comment by flp1990** [comment]
@gentaiscool thanks to reply,i have check #34 .i took your initial (except i set batch_size = 8,because  memory limit )setup to trainning  model on 960 hours librispeech datasets.After 3 days trainning(1 rtx titan )  it has obtain  Good results.
2020-11-13 09:32:47,406 - (Epoch 20) TRAIN LOSS:0.1903 CER:6.53% LR:0.0000167
2020-11-13 09:32:47,461 - VALID
2020-11-13 09:33:31,062 - VALID SET 0 LOSS:0.1780 CER:5.78%
2020-11-13 09:33:34,481 - SHUFFLE
2020-11-13 09:33:34,487 - TRAIN
2020-11-13 13:55:36,838 - (Epoch 21) TRAIN LOSS:0.1832 CER:6.25% LR:0.0000163
2020-11-13 13:55:36,894 - VALID
2020-11-13 13:56:31,427 - VALID SET 0 LOSS:0.1735 CER:5.59%
2020-11-13 13:56:35,347 - SHUFFLE
2020-11-13 13:56:35,349 - TRAIN
I want know your best results cer is   (transformer the batch_size bigger the results is better )?and  how many epochs  usually in train?thanks.

**Comment by gentaiscool** [comment]
I didn't record the results before. But, you can try 50-100 epochs. Btw, the CER showed in the log is not the autoregressive CER. You need to run the inference/test to check the real CER.

**Comment by flp1990** [comment]
Thanks, i will do it later :)

**Comment by Jingyilang** [comment]
> I didn't record the results before. But, you can try 50-100 epochs. Btw, the CER showed in the log is not the autoregressive CER. You need to run the inference/test to check the real CER.

What does "the CER showed in the log is not the autoregressive CER. You need to run the inference/test to check the real CER." mean?

**Comment by block98k** [comment]
> > I didn't record the results before. But, you can try 50-100 epochs. Btw, the CER showed in the log is not the autoregressive CER. You need to run the inference/test to check the real CER.
> 
> What does "the CER showed in the log is not the autoregressive CER. You need to run the inference/test to check the real CER." mean?

The CER during training is calculated with target-gt as input of decoder, so it must be higher than the true CER. You should run the test.py which gets the output one by one.

**Comment by uufitiygufd** [comment]
How to achieve training on librispeech database? I have been unable to complete the operation of downloading librispeech database. May I ask what modification should be made to the Librispeech.py file if I download the database by myself? In addition, after making changes to the database, what train instruction should I use to complete the training process? Thank you for your answer.
